### PR TITLE
The project is now platform agnostic and does not rely on any package manager.

### LINKED ISSUE #1: Making installation more portable.
Greetings,

Seeing that the installation script is currently hard coded for Arch, I'd like to propose setting up this project
under autotools in an attempt to create a simple, portable solution.

Would you be interested in such a contribution?

### PR DESCRIPTION
**Pull request by fvalasiad**
The installation script has been replaced by the
Makefile generated from the configure script.

To build from source tree:

autoreconf -i
./configure
make install

Or from a tarball release:
./configure
make install

You are going to need make, as well as autoconf and automake in case you choose the source tree installation.

Closes #1

### CODE REVIEW

**Comment by fvalasiad** [comment]
Make sure you add the new instructions in the project's README if you decide to merge.

File: configure.ac [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+if test "x$YT_DLP" = x; then
+    AC_MSG_ERROR([yt-dlp not found])
+fi
+
+AC_CHECK_PROG(MPV, mpv, [mpv])
+if test "x$MPV" = x; then
+    AC_MSG_ERROR([mpv not found])
+fi
+
+AC_CHECK_PROG(FF_BIN, firefox, [firefox])
```
Comment (MasterDevX) [inline configure.ac:35]: Overall, looks promising. The only issue I've found - there may be different types of the Firefox releases which have different executable names. For example, for classic Firefox release it's just "firefox", but for developer edition (which I personally use) it's "firefox-developer-edition", so for people who use non-classic Firefox, the dependency check will fail. Is there any way to check dependency by regex or something?

Also, as a person who never integrated autotools installation system into my projects, I'm a bit curious - are scripts under build-aux still needed even if all required autotools packages are installed in the system? Just want to keep the project clean and avoid having unnecessary files in the repo if there are any. Thanks!

Comment (fvalasiad) [reply configure.ac:35]: You could do:

```
AC_CHECK_PROG(FF_BIN, firefox, firefox-wayland, firefox-developer-edition, firefox-esr, ...)
// FF_BIN now contains the first program found in the list of comma separated executables
```

Which gives you a better control over what versions you want to allow since, for example, plain firefox won't allow you to install extensions not signed by mozilla.

So perhaps you'd want to be checking for other versions first:
```

AC_CHECK_PROG(FF_BIN, firefox-developer-edition, firefox-nightly, ..., firefox)

if "$FF_BIN" ~= "firefox*"
...
```

you are free to add commits to produce the exact behavior you are expecting.


Now regarding build-aux, it's standard practice to include it in the version control system, because:

> 
> 1. Portability: These auxiliary files help ensure that your project can be built and
>  configured consistently across different systems and environments. While
>  autotools themselves may be available, the version and configuration of
>  autotools can vary from system to system. The files in build-aux help bridge
>  these differences and provide a consistent build process.
> 
>   2. Customization: You might need to customize the build process or incorporate
>  specific features unique to your project. The files in build-aux can be modified
>  and extended to accommodate these requirements.
> 
>  3.  Reproducibility: Having these files in your project's repository ensures that
>  anyone who checks out the code can reproduce the build environment
>  precisely, even if they don't have the autotools installed on their system.

Basically, trying to respect the GNU license. I'd keep them, but it's up to you.

Comment (MasterDevX) [reply configure.ac:35]: Thanks for detailed explanation. I'll take a look at this a bit later today and looking forward to merge this with some changes. Well, because these files are automatically generated during the build process which is not really complex for this project, I'll probably exclude them from the repository so they won't confuse contributors who are unaware of how this build system works. Thanks for your contribution!
